frontend/test/metabase/lib/dom.unit.spec.js

```diff
@@ -95,3 +96,85 @@ describe("getUrlTarget", () => {
     expect(result).toBe("_blank");
   });
 });
+
+describe("open()", () => {
+  let getSdkGlobalPluginsMock;
+
+  beforeEach(async () => {
+    await mockIsEmbeddingSdk();
+    const sdkPluginsModule = await import(
+      "embedding-sdk-shared/lib/sdk-global-plugins"
+    );
+    getSdkGlobalPluginsMock = jest.spyOn(
+      sdkPluginsModule,
+      "getSdkGlobalPlugins",
+    );
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("should call handleLink plugin in the SDK", () => {
+    const handleLink = jest.fn().mockReturnValue(true);
+    getSdkGlobalPluginsMock.mockReturnValue({ handleLink });
+
+    const url = "https://example.com/dashboard/1";
+    open(url, {
+      openInSameWindow: jest.fn(),
+      openInBlankWindow: jest.fn(),
+    });
+
+    expect(handleLink).toHaveBeenCalledWith(url);
+  });
+
+  it("should prevent default behavior when handleLink returns true", () => {
+    const handleLink = jest.fn().mockReturnValue(true);
+    getSdkGlobalPluginsMock.mockReturnValue({ handleLink });
+
+    const openInSameWindow = jest.fn();
+    const openInBlankWindow = jest.fn();
+
+    const url = "https://example.com/dashboard/1";
+    open(url, {
+      openInSameWindow,
+      openInBlankWindow,
+    });
+
+    expect(handleLink).toHaveBeenCalledWith(url);
+    expect(openInSameWindow).not.toHaveBeenCalled();
+    expect(openInBlankWindow).not.toHaveBeenCalled();
+  });
+
+  it("should allow default behavior when handleLink returns false", () => {
```

Comment:
I'd also test when it's not returning anything e.g. most users might just left the function to return undefined when the condition didn't match their interests.